v2/apiserver/internal/core/jobs.go

```diff
@@ -643,6 +649,42 @@ func (j *jobsService) Cleanup(
 	return nil
 }
 
+func (j *jobsService) Timeout(
+	ctx context.Context,
+	eventID string,
+	jobName string,
+) error {
+	if err := j.authorize(ctx, RoleObserver, ""); err != nil {
+		return err
+	}
+
+	status, err := j.GetStatus(ctx, eventID, jobName)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"error retrieving status for event %q job %q",
+			eventID,
+			jobName,
+		)
+	}
+
+	// Update job status
+	now := time.Now()
+	status.Phase = JobPhaseTimedOut
+	status.Ended = &now
+
+	if err := j.UpdateStatus(ctx, eventID, jobName, status); err != nil {
+		return errors.Wrapf(
+			err,
+			"error updating status for event %q job %q",
+			eventID,
+			jobName,
+		)
+	}
+
+	return j.Cleanup(ctx, eventID, jobName)
```

Comment:
Non-blocking nit: There's room for a small optimization here. When you call `j.Cleanup(...)`, it's going to go back to the database to get the Event you've already got in hand. It's possible to refactor so that `j.Cleanup(...)` uses a private helper like `j.cleanup(...)` that accepts Event or Job, etc. as args. That helper could be invoked directly here to avoid the extra db call.

Again... this is non-blocking.

Reply:
Good call.  I'll add/update in https://github.com/brigadecore/brigade/pull/1419